test/unit/sdam/srv_polling.test.js

```diff
@@ -454,6 +439,33 @@ describe('Mongos SRV Polling', function () {
           new TopologyDescription(TopologyType.Sharded)
         )
       );
+      // This changes the topology to a sharded cluster
+      // We now have topology.s.handleSrvPolling assigned
+      expect(topology.s.handleSrvPolling).to.be.a('function');
+    });
+
+    it('two topologyDescriptionChanged events are emitted when receiving an SRV update with one server', function (done) {
+      let eventCount = 0;
+      topology.on(Topology.TOPOLOGY_DESCRIPTION_CHANGED, ev => {
+        if (eventCount === 0) {
+          // topologyDescriptionChanged event emitted from the handleSrvPolling function
+          expect(ev.topologyId).to.equal(1);
+          expect(ev.previousDescription.servers.size, 'previous should have 0 servers').to.equal(0);
+          expect(ev.newDescription.servers.size, 'next should have 1 server').to.equal(1);
+        }
+
+        if (eventCount >= 1) {
+          // Testing >= 1 to allow done to be called potentially more than once here catching erroneous events
```

Comment:
Since the actual expectations are the same in both blocks, you can probably pull them outside and update the "previous" expects to just use eventCount in place of 0 or 1, and just leave the `done()` call in the `if (eventCount >= 1)` block

Reply:
resolving with the update